### ISSUE TITLE
Istio IPv6 support 

### ISSUE THREAD

**Issue by wanghh2000** [istio/istio#13129]
**Describe the feature request**
Istio IPv6 support

**Describe alternatives you've considered**
Istio cannot work on IPv6

**Additional context**
Build a kubernetes cluster work on IPv6 and install istio 1.1.1.
After deploy sample bookinfo, update bookinfo code to listening on IPv6 address.
Still found istio does not work. 
Envoy support IPv6 by the following ticket.
https://github.com/envoyproxy/envoy/issues/351

**Comment by sanju2312** [comment]
Can you share what is the issue you are facing when using istio 1.1.1 with IPv6? Please share relevant log.

**Comment by wanghh2000** [comment]
The issue is that I deploy bookinfo in k8s cluster(use IPv6) installed istio 1.1.1 and cannot access bookinfo product page  by use command curl.
Success to access product page by curl when I deploy the same bookinfo without enable istio 1.1.1. 
Here are my test steps (enable istio):
1) Namespace istio-app 
[root@xxx /]# kubectl get ns -n istio-app --show-labels
NAME           STATUS   AGE     LABELS
cert-manager   Active   10d     
default        Active   10d     <none>
istio-app      Active   6d22h   istio-injection=enabled
istio-system   Active   10d     
[root@xxx /]#
2) Deploy bookinfo
[root@xxxx /]# kubectl get pod productpage-v1-645c595c59-b2649 -n istio-app -o wide
NAME                              READY   STATUS    RESTARTS   AGE     IP           NODE           NOMINATED NODE   READINESS GATES
productpage-v1-645c595c59-b2649   2/2     Running   0          3d22h   fd03::fcdb   xxxxx   <none>           <none>
[root@xxxx /]#
3) istio-ingress service
[root@xxxx /]# kubectl get svc istio-ingressgateway -n istio-system
NAME                   TYPE           CLUSTER-IP   EXTERNAL-IP   PORT(S)                                                                                                                                      AGE
istio-ingressgateway   LoadBalancer   fd02::9802   <pending>     80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:32253/TCP,15030:30058/TCP,15031:31464/TCP,15032:30524/TCP,15443:32208/TCP,15020:32455/TCP   6d23h
[root@xxxx /]#
4) Test productpage
[root@xxx /]# curl -g -6 'http://[fd03::fcdb]:9080/productpage'
curl: (7) Failed connect to fd03::fcdb:9080; Connection refused
[root@xxx /]#
[root@xxx /]# curl -g -6 'http://[fd02::9802]:80/productpage'
curl: (7) Failed connect to fd02::9802:80; Connection refused
[root@xxx /]#

[root@xxx /]# ping -6 fd03::fcdb
PING fd03::fcdb(fd03::fcdb) 56 data bytes
64 bytes from fd03::fcdb: icmp_seq=1 ttl=63 time=0.507 ms
64 bytes from fd03::fcdb: icmp_seq=2 ttl=63 time=0.446 ms
##############################################
Now I disable istio and retry the steps and it works.
[root@xxx /]# kubectl get pod productpage-v1-645c595c59-hjwdm  -n istio-app -o wide
NAME                              READY   STATUS    RESTARTS   AGE   IP           NODE           NOMINATED NODE   READINESS GATES
productpage-v1-645c595c59-hjwdm   1/1     Running   0          84s   fd03::fcdd   xxx   <none>           <none>
[root@xxx /]#
[root@xxx /]# curl -g -6 'http://[fd03::fcdd]:9080/productpage'
<!DOCTYPE html>
<html>
  <head>
    <title>Simple Bookstore App</title>
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
.......
 </body>
</html>
[root@xxx /]#
Canot not access productpage through istio-ingressgateway.
[root@xxx /]# kubectl get svc istio-ingressgateway -n istio-system
NAME                   TYPE           CLUSTER-IP   EXTERNAL-IP   PORT(S)                                                                                                                                      AGE
istio-ingressgateway   LoadBalancer   fd02::9802   <pending>     80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:32253/TCP,15030:30058/TCP,15031:31464/TCP,15032:30524/TCP,15443:32208/TCP,15020:32455/TCP   6d23h
[root@xxx /]# curl -g -6 'http://[fd02::9802]:80/productpage'
curl: (7) Failed connect to fd02::9802:80; Connection refused
[root@xxx /]#

**Comment by wanghh2000** [comment]
@sanju2312  Which pod log do you want to check? thanks.

**Comment by sbezverk** [comment]
@wanghh2000 Unfortunately 1.1 is still missing a couple of important changes, one of this has already been merged into master and second is still in need to merge state. Once it is done, ipv6 will work. I have istiov6 with knative working perfectly.

**Comment by sbezverk** [comment]
Good news, with merge of https://github.com/istio/istio/pull/13117 istio started fully supporting ipv6!!

```
istioctl proxy-config endpoints centos-ipv6-1-7cdd8bff7b-7vnjj.default
ENDPOINT                                         STATUS      CLUSTER
1.default.svc.k7.sbezverk.cisco.com
2001:470:b16e:1001:903a:3b30:d47a:bf02:53        HEALTHY     outbound|53||kube-dns.kube-system.svc.k7.sbezverk.cisco.com
2001:470:b16e:1001:903a:3b30:d47a:bf02:9153      HEALTHY     outbound|9153||kube-dns.kube-system.svc.k7.sbezverk.cisco.com
2001:470:b16e:1001:903a:3b30:d47a:bf03:53        HEALTHY     outbound|53||kube-dns.kube-system.svc.k7.sbezverk.cisco.com
2001:470:b16e:1001:903a:3b30:d47a:bf03:9153      HEALTHY     outbound|9153||kube-dns.kube-system.svc.k7.sbezverk.cisco.com
2001:470:b16e:1001:903a:3b30:d47a:bf2e:80        HEALTHY     outbound|80||istio-ingressgateway.istio-system.svc.k7.sbezverk.cisco.com
```

**Comment by mailzyok** [comment]
@sbezverk Thanks for your contribution. If i understand correctly, Istio now supports IPv4 only or IPv6 only network, the dual stack is not supported, right?

**Comment by sbezverk** [comment]
@mailzyok yep, that is right, k8s still does not support dual stack.

**Comment by wanghh2000** [comment]
@sbezverk Very glad to know news of istio started fully supporting ipv6.

**Comment by howardjohn** [comment]
IPv6 is working now (1.2 I think) so I am going to close this. More tracking under https://github.com/istio/istio/issues/1756
